scripts/check-c-api-docs.py

```diff
@@ -0,0 +1,93 @@
+"""
+A small script checking that all the C API functions are documented
+"""
+import os
+import sys
+
+from pycparser import c_ast, parse_file
+
+
+ROOT = os.path.join(os.path.dirname(__file__), "..")
+C_API_DOCS = os.path.join(ROOT, "docs", "src", "core", "reference", "c")
+FAKE_INCLUDES = os.path.join(ROOT, "python", "scripts", "include")
+METATENSOR_HEADER = os.path.relpath(
+    os.path.join(ROOT, "metatensor-core", "include", "metatensor.h")
+)
+
+
+ERRORS = 0
+
+
+def error(message):
+    global ERRORS
+    ERRORS += 1
+    print(message)
+
+
+def documented_functions():
+    functions = []
+
+    for root, _, paths in os.walk(C_API_DOCS):
+        for path in paths:
+            with open(os.path.join(root, path), encoding="utf8") as fd:
+                for line in fd:
+                    if line.startswith(".. doxygenfunction::"):
+                        name = line.split()[2]
+                        functions.append(name)
+
+    return functions
+
+
+def functions_in_outline():
+    MISC_FUNCTIONS = [
+        "mts_version",
+        "mts_last_error",
+        "mts_disable_panic_printing",
+        "mts_get_data_origin",
+        "mts_register_data_origin",
+        "mts_tensormap_load",
+        "mts_tensormap_load_buffer",
+        "mts_tensormap_save",
+        "mts_tensormap_save_buffer",
+    ]
+    functions = MISC_FUNCTIONS
```

Comment:
Maybe a comment that the misc function technically do not have an outline, because they their doc is structured differently, but we consider them as having one.

Reply:
done!